udata_front/theme/gouvfr/templates/reuse/list.html

```diff
@@ -39,40 +40,66 @@
+            class="fr-link fr-text--sm fr-m-0">
+            {{ _('Search in datasets') }}
+        </a>
     </div>
-    <div class="fr-grid-row fr-pt-9v justify-between align-items-center fr-grid-row--middle">
-        <h1 class="m-0 h2">{{ _('Reuses')}} <sup>{{reuses.total}}</sup></h1>
-        <a href="{{ url_for('datasets.list') }}" title="{{ _('Search in datasets') }}"
-            class="nav-link fs-sm">{{ _('Search in datasets') }}</a>
-    </div>
-    <div class="fr-grid-row fr-py-3w fr-displayed-lg">
-        <ul class="fr-tags-group">
-            <li>
-                <a
-                    href="{{ url_for('reuses.list') }}"
-                    class="fr-tag{% if not reuses.query._filters.topic %} active{% endif %}">
-                    {{ _('All') }}
-                </a>
-            </li>
-            {% for topic_id, label in reuses_topics %}
-                <li>
-                    <a
-                        href="{{ reuses.query.to_url(url, replace=True, **{'topic': topic_id}) }}"
-                        class="fr-tag{% if reuses.query._filters and reuses.query._filters.topic and reuses.query._filters.topic.topic == topic_id %} active{% endif %}"
-                    >
-                        {{ label }}
-                    </a>
-                </li>
-            {% endfor %}
-        </ul>
+    <div class="fr-pt-3v">
+        <form method="GET" class="fr-grid-row">
+            <div class="fr-search-bar fr-search-bar--lg w-100" role="search">
+                <label class="fr-label" for="search-reuses">
+                    {{ _('Search...') }}
+                </label>
+                <input
+                    name="q"
+                    class="fr-input"
+                    placeholder="{{ _('Search...') }}"
+                    type="search"
+                    id="search-reuses"
+                    value="{{reuses.query._query}}"
+                />
+                <button class="fr-btn" title="{{_('Search')}}" type="submit">
+                    {{ _('Search') }}
+                </button>
+            </div>
+            <div class="fr-grid-row fr-py-5v fr-displayed-lg">
+                <ul class="fr-tags-group">
+                    <li>
+                        <a
+                            href="{{ url_for('reuses.list') }}"
+                            class="fr-tag{% if not reuses.query._filters.topic %} active{% endif %}">
+                            {{ _('All') }}
+                        </a>
+                    </li>
+                    {% for topic_id, label in reuses_topics %}
+                        <li>
+                            <a
+                                href="{{ reuses.query.to_url(url, replace=True, **{'topic': topic_id}) }}"
+                                class="fr-tag{% if reuses.query._filters and reuses.query._filters.topic and reuses.query._filters.topic.topic == topic_id %} active{% endif %}"
```

Comment:
(to fix a bug introduced with search v2)

```suggestion
                                class="fr-tag{% if reuses.query._filters and reuses.query._filters.topic and reuses.query._filters.topic == topic_id %} active{% endif %}"
```